pallets/dapps-staking/src/pallet/mod.rs

```diff
@@ -916,16 +921,16 @@ pub mod pallet {
         ///
         /// This call can only be used during the pallet decommission process.
         #[pallet::call_index(14)]
-        #[pallet::weight(T::WeightInfo::claim_staker_without_restake())]
+        #[pallet::weight(T::WeightInfo::claim_staker_with_restake().max(T::WeightInfo::claim_staker_without_restake()))]
         pub fn claim_staker_for(
             origin: OriginFor<T>,
             staker: T::AccountId,
             contract_id: T::SmartContract,
         ) -> DispatchResultWithPostInfo {
             Self::ensure_pallet_enabled()?;
-            ensure_signed(origin)?;
+            let caller = ensure_signed(origin)?;
 
-            Self::internal_claim_staker_for(staker, contract_id)
+            Self::internal_claim_staker_for(caller, staker, contract_id, true)
```

Comment:
Should we ensure caller != staker ? Because it would be a great way to get the fee out of staking without waiting for unbunding period

Reply:
I was thinking of adding this check, but not because of the reason you mentioned. Great catch!